federation/schema_syncer.go

```diff
@@ -0,0 +1,84 @@
+package federation
+
+import (
+	"context"
+	"encoding/json"
+
+	"time"
+
+	"github.com/samsarahq/go/oops"
+	"github.com/samsarahq/thunder/graphql/introspection"
+)
+
+type SchemaSyncer interface {
```

Comment:
Add a comment for the interface. 
Add names to the input argument to make the interface easier to use.